compiler/src/iree/compiler/Dialect/HAL/IR/HALOps.cpp

```diff
@@ -864,6 +864,24 @@ Value AllocatorImportOp::getOperandSize(unsigned idx) { return {}; }
 
 Value AllocatorImportOp::getResultSize(unsigned idx) { return getLength(); }
 
+//===----------------------------------------------------------------------===//
+// hal.buffer.allocation.discard
+//===----------------------------------------------------------------------===//
+
+void BufferAllocationDiscardOp::getAsmResultNames(
+    function_ref<void(Value, StringRef)> setNameFn) {
+  setNameFn(getResult(), "was_terminal");
```

Comment:
why `was_terminal` as the result name?

Reply:
Good q - I don't think I thought too hard about it until just now :P This is a fetch-and-subtract operation and it's returning the value prior `is_terminal` before the subtract - so the value represents "was the buffer terminal at the time it was discarded" instead of "is the buffer terminal" (because that's after the subtract).